extensions/NV/SPV_NV_cluster_acceleration_structure.asciidoc

```diff
@@ -0,0 +1,226 @@
+
+Status
+------
+
+- Complete
+
+Version
+-------
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | 2025-01-01
+| Revision           | 1
+|========================================
+
+Dependencies
+------------
+
+This extension is written against the SPIR-V Specification,
+Version 1.6 Revision 1.
+
+This extension requires SPIR-V 1.4.
+
+This extension interacts with SPV_KHR_ray_tracing and SPV_KHR_ray_query.
+
+
+Overview
+--------
+
+This extension adds new functionality to support the Vulkan
+VK_NV_cluster_acceleration_structure extension in SPIR-V.
+
+
+Extension Name
+--------------
+
+To use this extension within a SPIR-V module, the following
+*OpExtension* must be present in the module:
+
+----
+OpExtension "SPV_NV_cluster_acceleration_structure"
+----
+
+
+
+New Capabilities
+----------------
+
+This extension introduces new capabilities:
+
+----
+RayTracingClusterAccelerationStructureNV
+----
+
+
+New Builtins
+------------
+
+Builtins added under the *RayTracingClusterAccelerationStructureNV* capability
+
+----
+ClusterIDNV
+----
+
+New Instructions
+----------------
+
+Instructions added under the *RayTracingClusterAccelerationStructureNV* capability
+
+----
+OpRayQueryGetIntersectionClusterIdNV
+OpHitObjectGetClusterIdNV
+----
+
+Modifications to the SPIR-V Specification, Version 1.1
+------------------------------------------------------
+
+(Modify Section 3.21, Builtin, adding rows to the Builtin table) ::
++
+--
+[cols="^1,10,^6",options="header",width = "100%"]
+|====
+2+^.^| Decoration | <<Capability,Enabling Capabilities>>
+| 5436 | *ClusterIDNV* +
+Provides triangle cluster ID of a hit with a triangle for the ray being traced, or -1
+if the hit object is not a cluster object. +
+Allowed in *AnyHitKHR* and *ClosestHitKHR* ray tracing execution models.
+|*RayTracingClusterAccelerationStructureNV*
+|====
+--
+
+
+(Modify Section 3.31, Capability, adding a row to the Capability table) ::
++
+--
+[cols="^.^5,25,15",options="header",width = "100%"]
+:capability: RayTracingKHR
+|====
+2+^.^| Capability | Implicitly Declares
+| 5437 | *RayTracingClusterAccelerationStructureNV* +
+| *RayTracingKHR*
+|====
+--
+
+(Add a new sub section 3.36.RQInstructions, Ray Query Instructions) ::
++
+--
+[cols="6*2"]
+|======
+5+|[[OpRayQueryGetIntersectionClusterIdNV]]*OpRayQueryGetIntersectionClusterIdNV* +
+ +
+ Returns the triangle cluster ID corresponding to the current intersection of the ray if the intersection is with a cluster object. Returns -1 otherwise. +
+ +
+ 'Result' is the returned cluster ID. +
+ +
+ 'Result Type' must be a 32-bit 'integer type' scalar. +
+ +
+ 'Intersection' must be the <id> of a 'constant instruction' with a 32-bit scalar 'integer type'. +
+ +
+ 'Intersection' identifies which intersection values should be returned for, either the current candidate or the
+ closest recorded hit so far; see {RayQueryUrl}#ray_query_intersection[Ray Query Intersection]. +
```

Comment:
FYI `{RayQueryUrl}#ray_query_intersection[Ray Query Intersection]` doesn't render correctly with github. I didn't check if asciidoctor processes it correctly, so really in case you missed it.

Reply:
Had missed adding RayQueryUrl 
Updated, should render correctly now